controllers/chat/group/groupAddMembers.js

```diff
@@ -61,7 +61,7 @@ const groupAddMembers = async (req, res) => {
   try {
     const responseChannel = await channelToAdd?.addMembers(filteredMembersObject);
 
-    channelMembers.map(async (member) => {
+    filteredMembersId.map(async (member) => {
       const targetUserModel = await UsersFunction.findUserById(User, member);
 
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `Array.prototype.map` with an async function could lead to unexpected behavior. The `map` function doesn't wait for promises to resolve before moving on to the next iteration. This means that all asynchronous operations are started almost simultaneously, and they may not complete in the order you expect. If the order of execution or completion is important, consider using a loop structure that supports asynchronous operations, like `for...of`.

```diff
-    filteredMembersId.map(async (member) => {
+    for (const member of filteredMembersId) {
       const targetUserModel = await UsersFunction.findUserById(User, member);
 
       const textOwnUser = `You added ${targetUserModel.username} to this group`;
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->